control/control.c

```diff
@@ -227,8 +229,37 @@ int main(int argc, char *argv[])
 		goto send_message;
 	}
 
+	if (!strcasecmp(command, "create")) {
+		has_response = true;
+		// need exactly one more argument (container config file)
+		if (optind != argc-1)
+			print_usage(argv[0]);
+
+		const char* cfgfile = argv[optind++];
+		off_t cfglen = file_size(cfgfile);
+		if (cfglen < 0) {
+			ERROR("Error accessing container config file %s.", cfgfile);
+			exit(-2);
+		}
+
+		unsigned char *cfg = mem_alloc(cfglen);
```

Comment:
free

Reply:
+ missing if null go to error, but not sure whether we check this in other places, so might not be necessary

Reply:
no free needed since exit() call on error syscall would return with error if mem_alloc failed.